SU2_CFD/src/solvers/CScalarSolver.cpp

```diff
@@ -645,12 +655,8 @@ void CScalarSolver::SetResidual_DualTime(CGeometry *geometry, CSolver **solver_c
       U_time_n  = nodes->GetSolution_time_n(iPoint);
       Density_n = 1.0;
 
-      //if (sst_model) {
-        if (incompressible)
-          Density_n = flowNodes->GetDensity(iPoint); // Temporary fix
-      //  else
-      //    Density_n = flowNodes->GetSolution_time_n(iPoint,0);
-      //}
+      if (incompressible)
+        Density_n = flowNodes->GetDensity(iPoint); 
```

Comment:
This is no tvalid for the SA model, the if sst needs to stay. ... but will be replaced in #1330

Reply:
But this piece of code is used in the transported scalar solver , not in the turbulence solver (it was copied from the turbulence solver). We should only care here about if the transported scalar is "implemented as" the SST model or the SA model. And it is implemented as the SST model because we have rho*Y just as we have rho*k and rho*omega.

Reply:
right... I am getting mixed-up with CScalarSolver over at #1330 because there it is general, my bad